src/store/utils.js

```diff
@@ -89,3 +89,202 @@ export const saveCurrentWeatherToLocalStorage = currentWeather => {
+    t01d: '15',
+    t02d: '15',
+    t03d: '15',
+    t04d: '15',
+    t05d: '15',
+    d01d: '18',
+    d02d: '18',
+    d03d: '18',
+    r01d: '18',
+    r02d: '18',
+    r03d: '18',
+    r04d: '18',
+    r05d: '12',
+    r06d: '12',
+    f01d: '19',
+    s01d: '22',
+    s02d: '22',
+    s03d: '22',
+    s04d: '29',
+    s05d: '25',
+    s06d: '19',
+    a01d: '11',
+    a02d: '11',
+    a03d: '5',
+    a04d: '11',
+    a05d: '11',
+    a06d: '11',
+    c01d: '1',
+    c02d: '2',
+    c03d: '4',
+    c04d: '7',
+    t01n: '15',
+    t02n: '15',
+    t03n: '15',
+    t04n: '15',
+    t05n: '15',
+    d01n: '18',
+    d02n: '18',
+    d03n: '18',
+    r01n: '18',
+    r02n: '18',
+    r03n: '18',
+    r04n: '18',
+    r05n: '12',
+    r06n: '12',
+    f01n: '19',
+    s01n: '22',
+    s02n: '22',
+    s03n: '22',
+    s04n: '29',
+    s05n: '25',
+    s06n: '19',
+    a01n: '11',
+    a02n: '11',
+    a03n: '5',
+    a04n: '11',
+    a05n: '11',
+    a06n: '11',
+    c01n: '33',
+    c02n: '34',
+    c03n: '36',
+    c04n: '7',
+  };
+  return iconsMapping[weatherbitIconCode] || weatherbitIconCode;
+};
+
+const formatTemperature = temperature => {
+  return temperature > 0 ? `+ ${Math.round(temperature)}` : Math.round(temperature);
+};
+
+const getDisplayStringByDate = date => {
+  const todayDate = new Date();
+  const tomorrowDate = new Date();
+  tomorrowDate.setDate(todayDate.getDate() + 1);
+  switch (date.getDay()) {
+    case todayDate.getDay():
+      return 'Сьогодні';
+    case tomorrowDate.getDay():
+      return 'Завтра';
+    default:
+      return date.toLocaleDateString('uk-UA', {
+        weekday: 'long',
+      });
+  }
+};
+
+const getAllDatesForHeader = jsonDataArray => {
+  const todayDate = new Date();
+  const tomorrowDate = new Date();
+  tomorrowDate.setDate(todayDate.getDate() + 1);
+
+  const datesArrayAsDate = jsonDataArray.map(arrayItem => {
+    const date = new Date(arrayItem.timestamp_local);
+    date.setHours(0);
+    return date;
+  });
+
+  const datesArrayInms = datesArrayAsDate.map(arrayItem => {
+    return arrayItem.valueOf();
+  });
+  return [...new Set(datesArrayInms)].map(dateItemInms => {
+    const dateItemAsDate = new Date(dateItemInms);
+    return {
+      displayString: getDisplayStringByDate(dateItemAsDate),
+      tooltipString:
+        dateItemAsDate.getDay() === todayDate.getDay() ||
+        dateItemAsDate.getDay() === tomorrowDate.getDay()
+          ? dateItemAsDate.toLocaleDateString('uk-UA', {
+              weekday: 'long',
+              day: '2-digit',
+              month: 'long',
+            })
+          : dateItemAsDate.toLocaleDateString('uk-UA', {
+              day: '2-digit',
+              month: 'long',
+            }),
+      columnsNumber: datesArrayAsDate.filter(date => date.getDay() === dateItemAsDate.getDay())
+        .length,
+    };
+  });
+};
+
+export const translateJSONToHourlyForecast = jsonResponse => {
+  const hourlyForecastData = {};
```

Comment:
I'm not sure if I got you... 
This function returns not only "data" array, but "datesWithColumnsNumber" array too. In one object.

Reply:
[Optional] 
I would like to you apply unix philosophy here. Make function that does one thing but does it well. It may sound silly but you know it works

Everything should be simple. It's so clear for me after years for programming.

Function with few logic concepts inside is much harder to maintain then few small functions, (one function per concept)

Reply:
Is this OK?